pkg/subscriptions/subscriptions.js

```diff
@@ -18,594 +18,104 @@
-              member: 'entitlement_status_changed'
-            },
-            function(path, dbus_interface, signal, args) {
-                update_timeout = null;
-                /*
-                 * status has changed, now get actual status via command line
-                 * since older versions of subscription-manager don't deliver this via DBus
-                 * note: subscription-manager needs superuser privileges
-                 */
-
-                /* if we're already waiting for a result, don't call again */
-                if (action_in_progress()) {
-                    status_update_requested = true;
-                    return;
-                }
-                get_subscription_status();
-
-            });
-
-        /* ideally we could get detailed subscription info via DBus, but we
-         * can't rely on this being present on all systems we work on
-         */
-        /*service.subscribe(
-            { path: "/EntitlementStatus",
-              interface: "org.freedesktop.DBUS.Properties",
-              member: "PropertiesChanged"
-            }, function(path, interface, signal, args) {
-            current.text('got properties');
-        });*/
-
-        $('#subscriptions-register').on('click', function() {
-            $('#subscriptions-register-dialog').modal('show');
-        });
-
-        /* once we have a proper DBus API, we can trigger an update via DBus - update_subscriptions()
-         * for now, read the relevant information directly to avoid delay
-         */
-        if (update_info) {
-            get_subscription_status();
-        }
-
-        return {
-            'register_system':      register_system
+        var dialogProps = {
+            'title': _("Register system"),
+            'body': React.createElement(subscriptionsRegister.dialogBody, registerDialogDetails),
         };
-    }
-
-    register_dialog_initialize();
-
-    function register_url_changed() {
-        var custom_url = $('#subscription-register-url-custom');
-        if (register_url === "default") {
-            custom_url.hide();
-        } else {
-            custom_url.show();
-            custom_url.focus();
-            custom_url.select();
-        }
-    }
-
-    function init(retrieve_current) {
-        if (retrieve_current === undefined)
-            retrieve_current = true;
-
-        $("#subscription-register-url").on("click", "li", function() {
-            register_url = $(this).attr("value");
-            $("#subscription-register-url button span").text($(this).text());
-            register_url_changed();
-        });
-
-        register_url = "default";
-        register_url_changed();
-        subscriptions.manager = subscription_manager(retrieve_current);
-    }
-
-    init();
+        if (renderDialog)
+            renderDialog(dialogProps, footerProps);
+        else
+            renderDialog = Dialog.show_modal_dialog(dialogProps, footerProps);
```

Comment:
> Why is this necessary here, but in other uses of the Dialog React class we don't need this?

The deciding factor here is where the decision to re-render comes from. If you change the state, React handles that internally. We pass in stuff via props though, so we need to manually trigger a new render. This is a convenience feature because the dialog pattern removes some "complexity in the middle".